src/components/button-icon/types.ts

```diff
@@ -0,0 +1,26 @@
+import { ReactElement } from "react";
+
+export enum BUTTON_ICON_KIND {
+  primary = "primary",
+  secondary = "secondary",
+  tertiary = "tertiary",
+  danger = "danger",
+}
+
+export enum BUTTON_ICON_SIZE {
```

Comment:
You can omit `mini` size of usual button, and use BUTTON_SIZE in order not to increase amount of enums